lib/src/client/call.dart

```diff
@@ -346,7 +346,17 @@ class ClientCall<Q, R> implements Response {
   /// If there's an error status then process it as a response error
   void _checkForErrorStatus(Map<String, String> metadata) {
     final status = metadata['grpc-status'];
-    final statusCode = int.parse(status ?? '0');
+    late final int statusCode;
+
+    if (status != null) {
+      statusCode = int.parse(status);
+    } else {
+      final httpStatus = metadata[':status'];
```

Comment:
The first line doesn’t compile with null safety, because `metadata.containsKey(':status')` does not ensure that `metadata[':status']` cannot return null and `int.tryParse` expects a not nullable String. I’ll just tweak it a little bit to make it work with null safety.
But I noticed the header validation breaks a lot of tests, because most of them don’t populate the headers with the `status` and the `content-type`. In addition I tried the header validation with our asp .net core server and the server doesn’t seem to send the content type, thus this change would break our application. 
Personally I would suggest to only check if the http code is present when validating the header.

Reply:
> In addition I tried the header validation with our asp .net core server and the server doesn’t seem to send the content type, thus this change would break our application.

Interesting. That's a violation of gRPC specifications. I see that https://github.com/grpc/grpc-dotnet seems to send back correct Content-Type, are you using some custom gRPC implementation for .NET? I also see that most native clients validate Content-Type they receive.

Reply:
I also looked at grpc-web implementation and it does validate content type it receives from the server: https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/grpcwebclientreadablestream.js#L138

Reply:
My bad, it was because of something else that the content-type was not sent in my case. Sorry for wasting your time.

Reply:
No worries! You are *not* wasting my time in any way. Thank *you* for spending time on this PR, this is really appreciated.